services/resource-usage-tracker/src/simcore_service_resource_usage_tracker/services/background_task_periodic_heartbeat_check.py

```diff
@@ -106,21 +107,39 @@ async def _close_unhealthy_service(
             running_service.last_heartbeat_at,
             running_service.pricing_unit_cost,
         )
+        _transaction_status = (
+            CreditTransactionStatus.NOT_BILLED
+            if running_service.service_type
+            == ResourceTrackerServiceType.COMPUTATIONAL_SERVICE
+            else CreditTransactionStatus.BILLED
+        )
         update_credit_transaction = CreditTransactionCreditsAndStatusUpdate(
             service_run_id=service_run_id,
             osparc_credits=make_negative(computed_credits),
-            transaction_status=(
-                CreditTransactionStatus.NOT_BILLED
-                if running_service.service_type
-                == ResourceTrackerServiceType.COMPUTATIONAL_SERVICE
-                else CreditTransactionStatus.BILLED
-            ),
+            transaction_status=_transaction_status,
         )
         await credit_transactions_db.update_credit_transaction_credits_and_status(
             db_engine, data=update_credit_transaction
         )
 
-    # 3. Release license seats in case some were checked out but not properly released.
+        # 3. If the credit transaction status is considered "NOT_BILLED", this might return
+        # the wallet to positive numbers. If, in the meantime, some transactions were marked as DEBT,
+        # we need to update them back to the BILLED state.
+        if _transaction_status == CreditTransactionStatus.NOT_BILLED:
+            wallet_total_credits = await credit_transactions_db.sum_wallet_credits(
+                db_engine,
+                product_name=running_service.product_name,
+                wallet_id=running_service.wallet_id,
+            )
+            if wallet_total_credits.available_osparc_credits >= 0:
+                await credit_transactions_db.batch_update_credit_transaction_status_for_in_debt_transactions(
+                    db_engine,
+                    project_id=None,
```

Comment:
why project_id None?

Reply:
Perhpas add a small comment? I wonder whether we should use more documented sentinels e.g. `ALL`